modules/storages/app/common/storages/peripherals/nextcloud_connection_validator.rb

```diff
@@ -70,29 +72,40 @@ def non_provisioned_user
                                       description: I18n.t("storages.health.connection_validation.oidc_non_provisioned_user")))
       end
 
+      def non_oidc_provisioned_user
+        return None() if @user.authentication_provider.is_a?(OpenIDConnect::Provider)
+
+        Some(ConnectionValidation.new(type: :warning,
+                                      error_code: :oidc_non_oidc_user,
+                                      timestamp: Time.current,
+                                      description: I18n.t("storages.health.connection_validation.oidc_non_oidc_user")))
+      end
+
       def token_usability
-        usable_token = @user.oidc_user_tokens.with_audience(@storage.audience).first
-        return None() if usable_token
+        service = OpenIDConnect::UserTokens::FetchService.new(user: @user)
+
+        result = service.access_token_for(audience: @storage.audience)
+        return None() if result.success?
+
+        error_code = case result.failure
+                     when :failed_refresh
```

Comment:
Comparing this `when` to the following `when`, I think the answer to my earlier question is: We should introduce a `token_refresh_response_invalid` error code as well.